### PR TITLE
EZP-27997: Provide support for URL fragment in UrlAlias router

### PR DESCRIPTION
**Pull request by webhdx**
> JIRA: https://jira.ez.no/browse/EZP-27997

# Description
Symfony 3.x introduced `_fragment` parameter in the default router which is used to add anchor to the URL. It is lacking in the `UrlAlias` router therefore it's not possible to generate friendly URL with anchor.

### CODE REVIEW

File: eZ/Publish/Core/MVC/Symfony/Routing/Generator/UrlAliasGenerator.php [hunk 1]
```diff
@@ -102,8 +102,12 @@ public function doGenerate($location, array $parameters)
             $rootLocationId = $this->rootLocationId;
         }
 
+        $anchor = array_key_exists('_fragment', $parameters)
```
Comment (lserwatka) [inline eZ/Publish/Core/MVC/Symfony/Routing/Generator/UrlAliasGenerator.php]: Can _fragment contain other things then # anchor? If so, then name can be confusing.

Comment (webhdx) [reply eZ/Publish/Core/MVC/Symfony/Routing/Generator/UrlAliasGenerator.php]: Fragment is probably better name than anchor: https://en.wikipedia.org/wiki/Fragment_identifier and it's also used by Symfony. I've just commited a variable name change: `$anchor -> $fragment`.

**Comment by alongosz** [comment]
Same comment as @adamwojs: please update unit test, AFAICS data provider update should be enough: `\eZ\Publish\Core\MVC\Symfony\Routing\Tests\UrlAliasGeneratorTest::providerTestDoGenerate`

**Comment by lserwatka** [comment]
@webhdx could you also create JIRA ticket for this change?

**Comment by webhdx** [comment]
Created JIRA ticket: https://jira.ez.no/browse/EZP-27997

cc @lserwatka

**Comment by webhdx** [comment]
@adamwojs @alongosz I've just added unit tests.

**Comment by alongosz** [comment]
@andrerom I thought about it also, but we would need to bump symfony in master from ^3.1 to ^3.2 as this is 3.2 feature AFAIK. Is it ok to do it? 
// cc @webhdx

**Comment by andrerom** [comment]
> we would need to bump symfony in master from ^3.1 to ^3.2 as this is 3.2 feature AFAIK. Is it ok to do it?


it is, as long the code continues to work on symfony 2.8

**Pull request closed without merging**